pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -217,6 +251,10 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
         _;
     }
 
+    /**
+     * @notice Add liquidity to a pool (including for the first time, known as "initializing" the pool).
```

Comment:
I would make the `@notice` tag more explicit that this is a hook.

Reply:
Noted on the exit as well.